File: frontend/src/types/scholarship.ts
export interface ScholarshipRecord {
  id: number
  student_id: number
  type: string
  name: string
  amount?: string
  status: 'pending' | 'approved' | 'rejected'
  semester: string
  note?: string
  created_at: string
  updated_at: string
}

export interface ScholarshipListResponse {
  data: ScholarshipRecord[]
  total: number
  page: number
  page_size: number
}

export const STATUS_MAP: Record<string, string> = {
  pending: '审核中',
  approved: '已发放',
  rejected: '已拒绝',
}

export const STATUS_TAG: Record<string, 'warning' | 'success' | 'danger'> = {
  pending: 'warning',
  approved: 'success',
  rejected: 'danger',
}

File: campus-service/app/schemas/scholarship.py
from datetime import datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, Field


class ScholarshipCreate(BaseModel):
    student_id: str = Field(..., min_length=1, max_length=32, description="学号")
    type: str = Field(..., max_length=64)
    name: str = Field(..., min_length=1, max_length=128)
    amount: Optional[Decimal] = Field(None, ge=0, max_digits=10, decimal_places=2)
    status: str = Field("pending", pattern=r"^(pending|approved|rejected)$")
    semester: str = Field("2025-2026-2", max_length=32)
    note: Optional[str] = None


class ScholarshipUpdate(BaseModel):
    student_id: Optional[str] = Field(None, min_length=1, max_length=32, description="学号")
    type: Optional[str] = Field(None, max_length=64)
    name: Optional[str] = Field(None, min_length=1, max_length=128)
    amount: Optional[Decimal] = Field(None, ge=0, max_digits=10, decimal_places=2)
    status: Optional[str] = Field(None, pattern=r"^(pending|approved|rejected)$")
    semester: Optional[str] = Field(None, max_length=32)
    note: Optional[str] = None


class ScholarshipOut(BaseModel):
    id: int
    student_id: int
    student_no: Optional[str] = None
    student_name: Optional[str] = None
    type: str
    name: str
    amount: Optional[Decimal]
    status: str
    semester: str
    note: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
